cpmpy/expressions/globalfunctions.py

```diff
@@ -379,31 +380,27 @@ def __init__(self, arr):
             raise ValueError("NValue takes an array as input")
         super().__init__("nvalue", arr)
 
-    def decompose_comparison(self, cmp_op, cpm_rhs):
+    def decompose(self):
         """
-        NValue(arr) can only be decomposed if it's part of a comparison
+        Decomposition of the Count constraint.
+        Returns
+        1) a numerical value to replace the constraint, and
+        2) a list of defining constraints, which should be enforced toplevel
 
         Based on "simple decomposition" from:
-        
+
             Bessiere, Christian, et al. "Decomposition of the NValue constraint."
             International Conference on Principles and Practice of Constraint Programming.
             Berlin, Heidelberg: Springer Berlin Heidelberg, 2010.
         """
-
         lbs, ubs = get_bounds(self.args)
         lb, ub = min(lbs), max(ubs)
 
-        constraints = []
-
-        # introduce boolvar for each possible value
-        bvars = boolvar(shape=(ub+1-lb,)) # shape is tuple to ensure it is a 1D array
+        n_values = 0
+        for v in range(lb, ub+1):
+            n_values += cp.any(a == v for a in self.args)
```

Comment:
its a bit of an unusual coding style
how about `return cp.sum(cp.any(self.args == v) for v in range(lb, ub+1))